### PR TITLE
Add command for reporting schema version

### PR DESCRIPTION
**Pull request by dboehmer**

### CODE REVIEW

File: lib/App/DH.pm [hunk 1]
```diff
@@ -261,15 +261,31 @@ sub cmd_upgrade {
   return;
 }
 
+=cmd version
+
+Report version of connection L</--connection_name>
```
Comment (kentfredric) [inline lib/App/DH.pm]: It should be noted that I don't think `->schema_version` queries the database to get the current database schema version.

I believe it does nothing more than query the logic in the relevant perl modules for the version of the _modules_ schema.

Hence why `write_ddl` uses that value as the "target version"

Perhaps this is grounds for a different set of commands, like `current_version` and `target_version` ?

However, implementing `current_version` is probably a lot of fun, and I'm kinda unsure what use `target_version` is.

File: lib/App/DH.pm [hunk 2]
```diff
@@ -261,15 +261,31 @@ sub cmd_upgrade {
...
+
+    dh.pl [...params] version
+
+=cut
+
+sub cmd_version {
```
Comment (kentfredric) [inline lib/App/DH.pm]: This should probably add an entry in the `=begin Pod::Coverage` block later in the source to appease the pod coverage tests.

**Review by kentfredric — CHANGES_REQUESTED** [review]

File: lib/App/DH.pm [hunk 3]
```diff
@@ -261,15 +261,31 @@ sub cmd_upgrade {
...
+  my $self = shift;
+  print $self->_dh->schema_version . "\n";
```
Comment (kentfredric) [inline lib/App/DH.pm:291]: Perl critic is complaining that this line is "un-checked", that is, the print could fail needing action. But I don't think any action is required here on failure, so a 

``` perl
## no critic (RequireCheckedSyscalls)
```

Should do the trick.

File: lib/App/DH.pm (hunk 1 above)
Comment (dboehmer) [reply lib/App/DH.pm]: I've now written `schema_version` and `database_version` which are the method names from DBIC::DeploymentHandler. Logic is already implemented there. Wasn't aware of the difference before.

**Comment by dboehmer** [comment]
Can you please review my 2nd commit in this PR?

File: lib/App/DH.pm [hunk 4]
```diff
@@ -281,6 +317,8 @@ my $list_cmds_usage =
     cmd_write_ddl
     cmd_install
     cmd_upgrade
+    cnd_database_version
```
Comment (kentfredric) [inline lib/App/DH.pm]: I suspect this should be "cmd" not "cnd"

Also, ideally these commits should be squashed down to a single commit, transient states without inherent value that never make it to a target aren't entirely useful and can be avoided :)

**Review by kentfredric — CHANGES_REQUESTED** [review]

File: lib/App/DH.pm (hunk 4 above)
Comment (dboehmer) [reply lib/App/DH.pm]: You're correct regarding `s/cnd/cmd/`.

I think there are valid arguments pro and against squashing the commits but as you requested I squashed down my PR to 1 commit.

I am not used to making PRs on Github that much. Do you get a notification once I update my repo? I was waiting for your last reply and then wrote a comment in order to gain your attention. I hope you don't mind.

Comment (kentfredric) [reply lib/App/DH.pm]: #### Squashing

Yeah. Its really a case-by-case thing. The individual commits themselves should be logical and atomic, and passing tests.

Squashing can be useful at this, but squashing is just a tool, and you have to use the right tool for the right job :)

Squashing unrelated changes for instance is obviously bad.

#### PRs.

No, I don't appear to get more emails for pushes. I thought it did, but I can't find any evidence it does.

So the only thing I really notice is when coveralls adds its bit after a travis run.

( But I don't really mind when people submit things that affect coverage, its not a hard metric, and I wish there was a way to disable the implication that reduced coverage is a failure ... )

**Review by kentfredric — APPROVED** [review]

**Pull request merged**